pkg/etcd/etcd.go

```diff
@@ -2047,7 +2047,10 @@ func snapshotRetention(retention int, snapshotPrefix string, snapshotDir string)
 		return nil
 	}
 	sort.Slice(snapshotFiles, func(i, j int) bool {
-		return snapshotFiles[i].Name() < snapshotFiles[j].Name()
+		// it takes the key from the file ex: etcd-snapshot-example-{date}, makes the split using "-" to find the date, takes the date and sort by date
+		stringI, stringJ := strings.Split(snapshotFiles[i].Name(), "-"), strings.Split(snapshotFiles[j].Name(), "-")
```

Comment:
Can you rename the variables to what they represent in terms of the snapshot names? We want to avoid Hungarian notation whenever possible in Go.

Reply:
Do you think is better to change the I and J for firstIndex and secondIndex?